bundles/org.openhab.binding.modbus.lambda/README.md

```diff
@@ -0,0 +1,174 @@
+| buffer1-error-number            | Number             | true      | [Nr]     | Buffer 1 Error Number (0 = No error)                                   |
+| buffer1-operating-state         | Number             | true      | [Nr]     | Buffer 1 Operating State: See Modbus description manual, link above    |
+| buffer1-actual-high-temperature | Number:Temperature | true      | [0.1 °C] | Actual temperature buffer high sensor				                   |
+| buffer1-actual-low-temperature  | Number:Temperature | true      | [0.1 °C] | Actual temperature buffer low sensor        						   |
+| Buffer 1 Maximum Temperature	  | Number:Temperature | false	   | [0.1 °C] | Setting for maximum buffer temperature (min = 25.0°C; max = 65.0°C)    |
+
+
+### Heating Circuit 1 Group
+
+This group contains general operational information about the heating circuit 1.
+
+| Channel ID                       				 | Item Type          | Read only | Unit         | Description                                               					   |
+| ---------------------------------------------- | -------------------| --------- | ------------ | --------------------------------------------------------------------------------|
+| heatingcircuit1-error-number      			 | Number             | true      | [Nr]         | Error Number (0 = No error)													   |
+| heatingcircuit1-operating-state            	 | Number             | true      | [Nr]         | Operating State: See Modbus description manual, link above|					   |
+| heatingcircuit1-flow-line-temperature   		 | Number:Temperature | true      | [Nr]         | Actual temperature flow line sensor         									   |
+| heatingcircuit1-return-line-temperature      	 | Number:Temperature | true      | [Nr]         | Actual temperature return line sensor         								   |
+| heatingcircuit1-room-device-temperature        | Number:Temperature | true      | [0.01°C]     | Actual temperature room device sensor (min = -29.9°C; max = 99.9°C) 	 		   |
+| heatingcircuit1-setpoint-flow-line-temperature | Number:Temperature | true      | [0.01°C]     | Setpoint temperature flow line (min = 15.0°C; max = 65.0°C)	   				   | 
+| heatingcircuit1-operating-mode  		   		 | Number 			  | true      | [Nr] 		 | Operating Mode: See Modbus description manual, link above    				   |
+| heatingcircuit1-offset-flow-line-temperature   | Number:Temperature | true      | [0.01°C]     | Setting for flow line temperature setpoint offset(min = -10.0K; max = 10.0K)	   |
+| heatingcircuit1-room-heating-temperature 		 | Number:Temperature | rue       | [0.01°C]     | Setting for heating mode room setpoint temperature(min = 15.0°C; max = 40.0 °C) |
+| eatingcircuit1-room-cooling-temperature        | Number:Temperature | true      | [0.01 l/min] | Setting for cooling mode room setpoint temperature(min = 15.0°C; max = 40.0 °C) |
+
+
+
+## Full Example
```

Comment:
I changed the tabs in my editor to spaces. 
I already supplied a sketchy example how to use the E-Manager function of the heat pump.

Reply:
As per template:
_Provide a full usage example based on **textual configuration files**._

Currently i see a yaml example from the UI.

Reply:
I started to split the bundle in seperate ones so that a user can use more than one boiler, buffer, heating circuit and heatpump. But I ran into problems with the handlers. I am not sure if that is a naming problem. 
Anyway I will not have time to correct your above mentioned issues within the next **two weeks**.

Anyway: Can you suggest one or two more modern bundles, preferably modbus, where I can learn how to improve my code, especially:
- 
When I split my bundle into 5 different ones, do I have to organize them in 5 different directories
openhab-addons\bundles\org.openhab.binding.modbus.lambdageneral\
openhab-addons\bundles\org.openhab.binding.modbus.lambdabuffer\
...
openhab-addons\bundles\org.openhab.binding.modbus.lambdaheatingcircuit\
or something like
openhab-addons\bundles\org.openhab.binding.modbus.lambdaheating\general\
...
openhab-addons\bundles\org.openhab.binding.modbus.lambdaheating\heatingcircuit\
with common code in 
openhab-addons\bundles\org.openhab.binding.modbus.lambdaheating\   ?

- handling of units;
- handlers;
- seperating parts of the Lambdahandler which is an instance (?) of BaseThingHandler into common usable code by the 5 different handlers I have to program.

Reply:
Ah, now understand your hesitation. We need to go back one step. There is no need to create seperate bundles. Just different Handlers within the bundle. No need to start fiddling with directory's, the change is much smaller.

1. Just copy paste the current `LambdaHandler.java` to `HeatpumpHandler.java`, `HeatingCircuitHandler.java` etc. Each Thing has its own handler. Obviously the contents of the files have to be stripped down to just the logic from that Thing it handles.
2. Adjust the `LambdaHandlerFactory.java` to return the right `Handler` for the right `Thing`
3. Re-arrange the thing-structure.xml so the rigth channels belong to the right Thing
4. Adjust README.md
5. Probably some improvements can be made with a `BaseThingHandler` to de-duplicate code in the handlers, but as being said that is implementation specific.

The astro binding uses this pattern (moon and sun are differnt things, they share a base handler) There are many more bindings using this pattenr, just open some and have q quick look at the handlers. The only thing that makes the modbus specific is the relation with the base modbus binding for the bridge, but that is not affecting this suggested pattern.
Hopefully this gives some guidance, i would suggest to start with just extract just one group e.g. the boiler (or pick one) and make that fully functional, if that is working fine we can just re-do the same to the other Things.

Reply:
I produced a version which provides seperate things for the main parts of the heatpump- that took some time.
I found a user (Agio Severin Corsten Foundation member) who tested it with two heating circuits, so I would like to commit the version to github. 

I corrected the problems in the README and the xml-files (at leasts I tried to).

Is there a way to erase all the old versions and submit only the new version to github?

You suggested:

> Provide a full usage example based on textual configuration files.

Where do I find these files? Do I just copy what I find in the UI with "Code"? That doesn't provide the format other READMEs provide.

As I understand it the configuration is stored in a Json-Database. Is there a way to extract the requested textual configuration from them?

Reply:
I found it! Instead of using the UI one can place files in $OPENHAB_CONF/things and $OPENHAB_CONF/items
Openhab creates these and shows them with a lock in the UI.